### ISSUE TITLE
Problem with colors of categories

### ISSUE THREAD

**Issue by logos88** [nbossard/packlist#58]
I have tried to use the categories field, but there is one color combination that it makes unreadable in my phone (Nexus 5 with LyneageOS)

![photo_2017-08-16_13-47-53](https://user-images.githubusercontent.com/4170088/29361918-e5003144-8289-11e7-808c-f9742afe1941.jpg)

As you can see, is the dark green last one.

**Comment by nbossard** [comment]
Correct this could be definitly improved, I wanted colors to be assigned automatically, and the color is currently assigned based on a checksum of category name... this needs to be improved.
What was the name of the category name unreadable?

**Comment by logos88** [comment]
The category name is: Neceser

**Comment by nbossard** [comment]
Thanks.
So the color is now quiet stupidly simple generated, see : 
`        // updating views
        if (curItem.getCategory() != null) {
            vHolderRecycle.global.setBackgroundColor(curItem.getCategory().hashCode());
        } else {
            vHolderRecycle.global.setBackgroundColor(0);
        }`
Improving it...

**Comment by nbossard** [comment]
Work is in progress

**Comment by nbossard** [comment]
Work is in progress : 

![screenshot_20170817-151445](https://user-images.githubusercontent.com/3626228/29414220-507e76f8-835f-11e7-8690-c4274b0b4424.png)

**Comment by logos88** [comment]
Great! That looks much better, thanks! : )

**Comment by nbossard** [comment]
This is still in branch   feature/iss58_improving_bgd_colors
as I think it is a bit too expensive too compute and needs to be computed only once for performance improvement and not for each line display.

**Comment by nbossard** [comment]
merged into develop, will be in next release

**Comment by nbossard** [comment]
now available in production, closing issue
